src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/SafeHandles.cs

```diff
@@ -114,34 +114,40 @@ public static void DeleteKeyContainer(SafeCertContextHandle pCertContext)
                 {
                     CRYPT_KEY_PROV_INFO* pProvInfo = (CRYPT_KEY_PROV_INFO*)pProvInfoAsBytes;
 
-                    // For UWP the key was opened in a CNG-only context, so it can/should be deleted via CngKey.Delete.
-                    // In all other contexts, the key was loaded into CAPI, so deleting it via CryptAcquireContext is
-                    // the correct action.
-#if NETNATIVE
-                    string providerName = Marshal.PtrToStringUni((IntPtr)(pProvInfo->pwszProvName));
-                    string keyContainerName = Marshal.PtrToStringUni((IntPtr)(pProvInfo->pwszContainerName));
-
-                    try
+                    if (pProvInfo->dwProvType == 0)
```

Comment:
This is probably worth a comment along the lines of "When opening the key the dwProvType==0 check isn't correct, because most CAPI keys can be loaded in CNG. But for deleting the key handle this check is correct, because it's asking about the storage of the key, not the functionality of it".

Because I was in the middle of asking why it was valid here.  (The comment in the if doesn't quite get it, since it talks about how the key was opened)